packages/vgplot/src/spec/to-module.js

```diff
@@ -162,8 +164,9 @@ class CodegenContext extends ParseContext {
   }
 }
 
-function parseExpression(spec, ctx) {
-  const { expr, label } = spec;
+function parseExprOrAgg(spec, ctx, key, func_name) {
```

Comment:
Instead, maybe rewrite parseExpression like so:

```js
function parseExpression(spec, ctx, key = 'expr', method = 'sql') {
  const { label } = spec;
  const expr = spec[key];
  const tokens = expr.split(/(\\'|\\"|"(?:\\"|[^"])*"|'(?:\\'|[^'])*'|\$\w+)/g);
  let str = '';

  for (let i = 0; i < tokens.length; ++i) {
    const tok = tokens[i];
    if (tok.startsWith('$')) {
      str += `\${${ctx.maybeParam(tok)}}`;
    } else {
      str += tok;
    }
  }

  return `vg.${method}\`${str}\``
    + (label ? `.annotate({ label: ${JSON.stringify(label)} })` : '');
}
```

To parse an aggregation we then only need to call parseExpression with the second and third arguments both equal to `'agg'`.